infra/gravity/provision.go

```diff
@@ -57,8 +68,152 @@ func configureVMs(baseCtx context.Context, log logrus.FieldLogger, params cloudD
 	return sorted(gravityNodes), nil
 }
 
-// Provision gets VMs up, running and ready to use
+// Provision will attempt to provision the requested cluster
 func (c *TestContext) Provision(cfg ProvisionerConfig) ([]Gravity, DestroyFn, error) {
+	// store the configuration used for provisioning
+	c.provisionerCfg = cfg
+
+	var nodes []Gravity
+	var destroy DestroyFn
+	var err error
+	switch cfg.CloudProvider {
+	case "azure", "aws":
+		nodes, destroy, err = c.provisionCloud(cfg)
+	case "ops":
+		nodes, destroy, err = c.provisionOps(cfg)
+	default:
+		err = trace.BadParameter("unkown cloud provider: %v", cfg.CloudProvider)
+	}
+
+	// call `destroyFn` if provided to destroy infrastructure
+	if err != nil && destroy != nil {
+		dErr := destroy()
+		if dErr != nil {
+			err = trace.NewAggregate(err, dErr)
+		}
+		// if we destroy the cluster, make sure nodes and the destroy functions are returned as nil
+		nodes = nil
+		destroy = nil
+	}
+
+	return nodes, destroy, trace.Wrap(err)
+}
+
+// provisionOps utilizes an ops center installation flow to complete cluster installation
+func (c *TestContext) provisionOps(cfg ProvisionerConfig) ([]Gravity, DestroyFn, error) {
+	c.Logger().WithField("config", cfg).Debug("Provisioning via Ops Center")
+
+	// verify connection before starting provisioning
+	c.Logger().Debug("attempting to connect to AWS api")
+	sess, err := session.NewSession(&aws.Config{
+		Region:      aws.String(cfg.Ops.EC2Region),
+		Credentials: credentials.NewStaticCredentials(cfg.Ops.EC2AccessKey, cfg.Ops.EC2SecretKey, ""),
+	})
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+	c.Logger().Debug("logging into the ops center")
+	out, err := exec.Command("tele", "login", "-o", cfg.Ops.URL, "--key", cfg.Ops.OpsKey).CombinedOutput()
+	if err != nil {
+		return nil, nil, trace.WrapWithMessage(err, string(out))
+	}
+
+	// generate a random cluster name
+	clusterName := fmt.Sprint("robotest-", uuid.NewV4().String())
```

Comment:
would be helpful to add test case name prefix to make it easier to identify the tests